ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -796,10 +796,15 @@ func (r *RayClusterReconciler) reconcilePods(ctx context.Context, instance *rayv
 			}
 		}
 		// A replica can contain multiple hosts, so we need to calculate this based on the number of hosts per replica.
+		// If the user doesn't install the CRD with `NumOfHosts`, the zero value of `NumOfHosts`, which is 0, will be used.
+		// Hence, all workers will be deleted. Here, we set `NumOfHosts` to max(1, `NumOfHosts`) to avoid this situation.
+		if worker.NumOfHosts <= 0 {
```

Comment:
In hindsight, `NumOfHosts` should have been a pointer so we can check for presence instead of using it's zero value

Reply:
Makes sense. At that time, I didn't take the mismatch between the CRD and the KubeRay image into consideration when reviewing the PR. Do you think we need to update the CRD in this PR? I think it will still be compatible with the YAML files for v1.1.0.

Reply:
Hmm.. hard to say. It won't break any YAML but it would break Go references to the types. 

I think we should also update the upgrade guidance to always upgrade the CRD before the binary. I can try to update the docs for this.

Reply:
Thanks!

Reply:
https://github.com/ray-project/ray/pull/44826

Reply:
Confirmed this bug doesn't occur when I upgrade with these two steps:
```
$ kubectl apply --server-side --force-conflicts -k "github.com/ray-project/kuberay/ray-operator/config/crd?ref=v1.1.0"
$ helm upgrade kuberay-operator kuberay/kuberay-operator --version v1.1.0
```